unitTests/ui/ResponsiveComponents/ResponsiveComponentsManagerTest.ts

```diff
@@ -34,22 +39,55 @@ export function ResponsiveComponentsManagerTest() {
       jasmine.clock().uninstall();
     });
 
-    it('calls handle resize event when resize listener is called', () => {
-      root.width = () => 400;
-      responsiveComponentsManager.register(responsiveComponent, root, 'id', component, {});
-      responsiveComponentsManager.resizeListener();
-      jasmine.clock().tick(250);
-      expect(handleResizeEvent).toHaveBeenCalled();
+    describe('when the search interface responsive mode is set to *auto*', () => {
+      beforeEach(() => {
+        setResponsiveMode('auto');
+      });
+
+      it('should calls handle resize event when resize listener is called', () => {
+        root.width = () => 400;
+        responsiveComponentsManager.register(responsiveComponent, root, 'id', component, {});
+        responsiveComponentsManager.resizeListener();
+        jasmine.clock().tick(250);
+        expect(handleResizeEvent).toHaveBeenCalled();
+      });
+
+      describe('and the root element width is zero', () => {
+        beforeEach(() => {
+          root.width = () => 0;
+        });
+
+        it('should  not calls handle resize event when resize listener is called', () => {
+          responsiveComponentsManager.register(responsiveComponent, root, 'id', component, {});
+          responsiveComponentsManager.resizeListener();
+          jasmine.clock().tick(250);
+          expect(handleResizeEvent).not.toHaveBeenCalled();
+        });
+
+        it('should  not calls handle resize event when it registers component', () => {
+          responsiveComponentsManager.register(responsiveComponent, root, 'id', component, {});
+          expect(handleResizeEvent).not.toHaveBeenCalled();
+        });
+      });
     });
 
-    it('does not calls handle resize event when resize listener is called and width is zero', () => {
-      root.width = () => 0;
-      responsiveComponentsManager.register(responsiveComponent, root, 'id', component, {});
-      jasmine.clock().tick(250);
+    describe('when the search interface responsive mode is not set to *auto*', () => {
+      beforeEach(() => {
+        setResponsiveMode('small');
```

Comment:
Rather than hard-coding the mode `small`, would it be possible to reference it?

Reply:
sure